src/Raven.Server/Documents/TimeSeries/TimeSeriesStorage.cs

```diff
@@ -200,8 +216,19 @@ private long PurgeSegments(long upto, DocumentsOperationContext context, Collect
                 {
                     pendingDeletion.MultiDelete(collectionName.Name, etagSlice);
                 }
-
                 outdated.Clear();
+
+                foreach (var tsKey in uniqueTimeSeries)
+                {
+                    if (table.SeekOnePrimaryKeyWithPrefix(tsKey, Slices.BeforeAllKeys, out _) == false)
+                    {
+                        using (Slice.External(context.Allocator, tsKey, tsKey.Size - 1, out var statsKey))
```

Comment:
Why `- 1` ?

Reply:
the key slice is `docId/tsName/` and I need to transform it into `docId/tsName` 
note: the `/` represents the record separator char